### PR TITLE
[RFC] docs: Remove references to Windows 3.1

### PR DESCRIPTION
**Pull request by sethjackson**
We don't support Windows 3.1 anymore.

### CODE REVIEW

**Comment by sethjackson** [comment]
Travis has been flaky today. Not sure why this failed.

**Comment by Pyrohh** [comment]
Should probably remove this too: https://github.com/sethjackson/neovim/blob/windows-doc-cleanup/runtime/doc/os_win32.txt#L214-L224

File: runtime/doc/gui_w32.txt [hunk 1]
```diff
@@ -48,9 +48,6 @@ vimrc or gvimrc file: >
 	au GUIEnter * simalt ~x
 <
 								*gui-w32s*
```
Comment (Pyrohh) [inline runtime/doc/gui_w32.txt]: Since the text below is being removed, the same should be done for this tag.

**Comment by sethjackson** [comment]
Thanks @Pyrohh. I think I removed the tag correctly? Sadly I'm not too familiar with the Vim doc format.

**Comment by Pyrohh** [comment]
> I think I removed the tag correctly now? Sadly I'm not too familiar with the Vim doc format.

Yes, it seems fine to me. The Vim help format is actually really simple; this text from `:help help-writing` should explain everything there is to know about tags:

```
To define a help tag, place the name between asterisks (*tag-name*).  The
tag-name should be different from all the Vim help tag names and ideally
should begin with the name of the Vim plugin.  The tag name is usually right
aligned on a line.

When referring to an existing help tag and to create a hot-link, place the
name between two bars (|) eg. |help-writing|.
```

**Comment by Pyrohh** [comment]
Oh, and I committed this with 2 extra changes in `gui_w32.txt`, thanks.

**Comment by sethjackson** [comment]
Didn't know about `:help help-writing`. Thanks!

**Pull request closed without merging**